lib/far_mar/vendor.rb

```diff
@@ -0,0 +1,71 @@
+class FarMar::Vendor < FarMar::Base
+  attr_accessor :id, :name, :no_of_employees, :market_id
+  FILE = ("support/vendors.csv")
+
+  def initialize(attrs)
+   @id= attrs[0].to_i
+   @name = attrs[1]
+   @no_of_employees = attrs[2].to_i
+   @market_id = attrs[3].to_i
+  end
+
+
+  def market
+    markets = FarMar::Market.all
+    all_markets = markets.group_by do |obj|
+      obj.id
+    end
+    all_markets[@id].first
```

Comment:
This should be `all_markets[@market_id]`, because it's grouping `all_markets` by market id not vendor id.